src/buffers/RecordingBuffer.cpp

```diff
@@ -42,3 +42,58 @@ time_t RecordingBuffer::GetBufferEndTime()
 {
   return Buffer::GetEndTime();
 }
+
+int RecordingBuffer::Duration(void)
+{
+  if (m_isRecording)
+  {
+    time_t endTime =  time(nullptr);
+    int diff = (int) (endTime - m_startTime - 10);
+    if (diff > 0)
+    {
+      return diff;
+    }
+    else
+    {
+      return 0;
+    }
+  }
+  else
+  {
+    return m_Duration;
+  }
+}
+bool RecordingBuffer::Open(const std::string inputUrl,const PVR_RECORDING &recording)
+{
+  m_Duration = recording.iDuration;
+  if (recording.iDuration + recording.recordingTime > time(nullptr))
+  {
+    m_startTime = recording.recordingTime;
+    XBMC->Log(LOG_DEBUG, "RecordingBuffer::Open In Progress %d %lld", recording.iDuration, recording.recordingTime);
+    m_isRecording = TRUE;
```

Comment:
Checks are failing right now, because of these.  use true and false, not TRUE and FALSE in C++